kvmagent/kvmagent/plugins/vm_local_volume_cache/command_wrapper/qemu_img.py

```
@@ -0,0 +1,362 @@
+            ch = stdout_stream.read(1)
+            if not ch:
+                # EOF, process finished                    
+                if buf.strip():
+                    try:
+                        progress = float(buf.strip().replace("(", "").replace(")", "").split("/").pop(0))
+                        yield progress
+                    except (ValueError, IndexError):
+                        pass
+                stderr = stderr_stream.read()
+                if stderr:
+                   raise Exception("failed to flush qcow2 image, stderr: %s" % stderr)
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)